tour_model_eval/get_request_percentage.py

```diff
@@ -0,0 +1,131 @@
+def match_day(trip,bin,filter_trips):
+    if bin:
+        t = filter_trips[bin[0]]
+        if trip['data']['start_local_dt']['year']==t['data']['start_local_dt']['year']\
+                and trip['data']['start_local_dt']['month']==t['data']['start_local_dt']['month']\
+                and trip['data']['start_local_dt']['day']==t['data']['start_local_dt']['day']:
+            return True
+    return False
+
+
+# This function is to compare a trip with a group of trips to see if they happened in a same month
+def match_month(trip,bin,filter_trips):
+    if bin:
+        t = filter_trips[bin[0]]
+        if trip['data']['start_local_dt']['year']==t['data']['start_local_dt']['year']\
+                and trip['data']['start_local_dt']['month']==t['data']['start_local_dt']['month']:
+            return True
+    return False
+
+
+# This function bins trips according to ['start_local_dt']
+def bin_date(trip_ls,filter_trips,day=None,month=None):
+    bin_date = []
+    for trip_index in trip_ls:
+        added = False
+        trip = filter_trips[trip_index]
+
+        for bin in bin_date:
+            if day:
+                if match_day(trip,bin,filter_trips):
+                    bin.append(trip_index)
+                    added = True
+                    break
+            if month:
+                if match_month(trip,bin,filter_trips):
+                    bin.append(trip_index)
+                    added = True
+                    break
+
+        if not added:
+            bin_date.append([trip_index])
+
+    return bin_date
+
+
+def find_first_trip(filter_trips,bin):
+    trip_ts = [filter_trips[trip_idx]['data']["start_ts"] for trip_idx in bin]
+    # - early_idx_in_bin: the earliest trip position in the bin
+    # ts = [20,10,40,5,100]
+    # early_idx_in_bin = 3
+    # early trip_index = 5
+    early_idx_in_bin = trip_ts.index(min(trip_ts))
+    # - early_trip_index: the original index of the earliest trip
+    early_trip_index = bin[early_idx_in_bin]
+    return early_trip_index, early_idx_in_bin
+
+
+# collect requested trips and common trips(no need to request) indices above cutoff
+def requested_trips_ab_cutoff(new_bins, filter_trips):
+    # collect requested trip indices above cutoff
+    ab_trip_ls = []
+    # collect common trip indices above cutoff
+    no_req_trip_ls = []
+    for bin in new_bins:
+        early_trip_index, early_idx_in_bin = find_first_trip(filter_trips, bin)
+        ab_trip_ls.append(early_trip_index)
+
+        # The following loop collects the original indices of the rest of the trips in the bin. Since they are not the
+        # earliest one, we don't need to request for user labels
+        # >>> x = [100,200,300]
+        # >>> x.remove(100); x
+        # [200, 300]
+        no_req_trip_subls = copy.copy(bin)
+        no_req_trip_subls.remove(early_trip_index)
+        no_req_trip_ls.append(no_req_trip_subls)
+    no_req_trip_ls = list(itertools.chain(*no_req_trip_ls))
```

Comment:
you can do the following to keep the code similar to `ab_trip_ls`, right?

```suggestion
        # >>> x = [1,2,3]
        # >>> x.extend([4,5,6]); x
        # [1, 2, 3, 4, 5, 6]
        no_req_trip_ls.extend(no_req_trip_subls)
```